docs/common/transactions.md

```diff
@@ -0,0 +1,72 @@
+## API
+
+If you want to execute multiple statements under one transaction, use `transaction` function. 
+
+```kotlin
+val players = listOf<Player>()
+database.playerQueries.transaction {
+  players.forEach { player ->
+    database.playerQueries.insert(
+      player_number = player.number,
+      full_name = player.fullName
+    )
+  }
+}
+```
+
+To return a value from a transaction, use the `transactionWithReturn` function.
+
+```kotlin
+val players = database.playerQueries.transactionWithReturn {
+  database.playerQueries.selectAll().executeAsList()
+}
+```
+
+## Rollback
+
+Transactions will roll back if an exception occurs anywhere in them. You can manually roll back a
+transaction anywhere inside of one, but if your transaction returns a value you will need to specify
+a value for the transaction to return.
+
+```kotlin
+database.playerQueries.transaction {
+  players.forEach { player ->
+    if (player.number == 0) rollback()
+    database.playerQueries.insert(
+      player_number = player.number,
+      full_name = player.fullName
+    )
+  }
+}
+```
+
+```kotlin
+val numberInserted = database.playerQueries.transaction {
```

Comment:
Sorry for chiming in, was just browsing the docs out of curiosity - I guess this one should be `transactionWithReturn`  right?

Reply:
thank you! If you review that is super helpful 🙇 🙇 🙇 🙇